tethysext/atcore/tests/unit_tests/services/map_manager.py

```diff
@@ -1445,3 +1447,28 @@ def test_build_legend_with_color_ramp(self, mock_gccrd):
 
         ret = map_manager.build_legend(mock_layer, units='Ft')
         self.assertEqual(ret, expected)
+
+    def test_build_legend_with_use_geoserver_legend(self):
+        mock_layer = {
+            'use_geoserver_legend': True, 'endpoint': 'http://localhost:8181/geoserver/wms',
+            'layer_name': 'test:layer_name', 'layer_title': 'Test_Title', 'layer_variable': 'test:layer_variable',
+            'layer_id': ''
+        }
+
+        expected = {
+            'is_geoserver_legend': True,
+            'legend_id': 'test_layer_variable',
+            'title': 'Test Title',
+            'legend_url': (
```

Comment:
Instead of providing a `legend_url` parameter, we should build it dynamically from the information they give us for the layer (`endpoint`). It should be something like: 
`legend_url = f'{endpoint}?service=WMS&version=1.1.0&request=GetLegendGraphic&layer=test:layer_name&format=image/png'`
Then add a `legend_options` argument that takes a dictionary for things like `hideEmptyRules` and `transparent` options.